### PR TITLE
Readme: Add PyPI version badge

### PR DESCRIPTION
**Pull request by EwoutH**
Add PyPI version badge with an URL to the PyPI to quickly see what the latest version of mpi4py on PyPI is.

[PyPi](https://pypi.python.org/pypi/mpi4py)

### CODE REVIEW

**Comment by dalcinl** [comment]
I have mixed feeling about this. All the current badges related to development. For example, the CI badges show the state of the master branch, the ReadTheDocs badge link to the documentation from the master branch. The right side pane of the GitHub UI currently shows the latest 4.0.0 in the "Releases" section.

Perhaps what we need is a separate line of badges AFTER the current line of badges, with with links to places where to get mpi4py. The first and most obvious one is the pypi badge you added here, but maybe we should somehow add a conda-forge badge, and links to https://anaconda.org/mpi4py. 

pypi 
wheel 
conda-forge
homebrew

Also maybe Linux distros?

fedora
debian
ubuntu
archlinux

**Comment by EwoutH** [comment]
I split the badges into two new lines, one with the package versions on general Python package distribution platforms (PyPI and (ana)conda), and a line with OS specific distribution versions (Homebrew and Linux distros).

File: README.rst [hunk 1]
```diff
@@ -2,20 +2,38 @@
...
-   :target: https://dev.azure.com/mpi4py/mpi4py/_build
-.. image::  https://ci.appveyor.com/api/projects/status/whh5xovp217h0f7n?svg=true
-   :target: https://ci.appveyor.com/project/mpi4py/mpi4py
-.. image::  https://circleci.com/gh/mpi4py/mpi4py.svg?style=shield
-   :target: https://circleci.com/gh/mpi4py/mpi4py
-.. image::  https://codecov.io/gh/mpi4py/mpi4py/branch/master/graph/badge.svg
-   :target: https://codecov.io/gh/mpi4py/mpi4py
-.. image::  https://scan.coverity.com/projects/mpi4py-mpi4py/badge.svg
-   :target: https://scan.coverity.com/projects/mpi4py-mpi4py
+-  .. image:: https://github.com/mpi4py/mpi4py/workflows/ci/badge.svg?branch=master
```
Comment (dalcinl) [inline README.rst:5]: I don't quite like using list, the bullet point followed by badges looks a bit awful IMO. 
What if you just add a blank line as a separator? That should be enough to force the related badges on its own line.

Comment (EwoutH) [reply README.rst:5]: Tried that, but unfortunately that didn't seem to work with RST formatting.

**Comment by dalcinl** [comment]
@EwoutH Take a look here: https://github.com/mpi4py/mpi4py/tree/readme-badges
Does it look good enough? Do you have any strong objections?

**Comment by EwoutH** [comment]
Looks really nice! I thought I tried the same thing, but apparently not properly.

Shall I close this PR?

**Comment by dalcinl** [comment]
Yea, let's close this PR, I'll merge my branch manually.

**Pull request closed without merging**